miniapps/common/CMakeLists.txt

```diff
@@ -27,3 +27,6 @@ endif()
 
 add_library(mfem_miniapps_common ${MFEM_MINIAPPS_COMMON_SOURCES}
             ${MFEM_MINIAPPS_COMMON_HEADERS})
+if (MFEM_USE_PUMI)
```

Comment:
Unless there are changes in `miniapps/`, this shouldn't be necessary.

Reply:
OK.  We'll remove this.

Reply:
Removing

```
if (MFEM_USE_PUMI)
 target_link_libraries(mfem_miniapps_common PUBLIC SCOREC::core)
endif()
```

 results in a compilation error:

```
[ 32%] Building CXX object CMakeFiles/mfem.dir/fem/hybridization.cpp.o
In file included from /lore/cwsmith/develop/mfem/mfem/c:40:0,
                from /lore/cwsmith/develop/mfem/mfem/mfem.hpp:38,
                from /lore/cwsmith/develop/mfem/mfem-build/mfem.hpp:3,
                from /lore/cwsmith/develop/mfem/mfem/miniapps/common/fem_extras.hpp:15,
                from /lore/cwsmith/develop/mfem/mfem/miniapps/common/fem_extras.cpp:12:
/lore/cwsmith/develop/mfem/mfem/mesh/pumi.hpp:39:18: fatal error: pumi.h: No such file or directory
#include <pumi.h>
                 ^
compilation terminated.
```

I don't think we can remove this `target_link_libraries` (which brings in the compiler flag to include the pumi install directory) call without changes to where the pumi headers are included.

Reply:
@cwsmith, it looks like when `MFEM_USE_PUMI` is enabled, `SCOREC::core` needs to be a dependency of the library, `mfem`. That way the dependency is propagated to everything that depends on `mfem`.

Reply:
@v-dobrev Thank you.  SCOREC::core was already set as a dependency of mfem:
https://github.com/mfem/mfem/blob/1bd99fd5dc0541e60c081f4330e61788d9b0bdb5/CMakeLists.txt#L312

but, digging a bit more, and it looks like `mfem_miniapps_common` does not depend on `mfem`.  Removing:
https://github.com/mfem/mfem/blob/1bd99fd5dc0541e60c081f4330e61788d9b0bdb5/miniapps/common/CMakeLists.txt#L30-L32
and adding the following line
```
target_link_libraries(mfem_miniapps_common PUBLIC mfem)
```
gets me through the compile.  I'm not sure if this is a cleaner solution though.

Reply:
OK, pick the one that you prefer.

Reply:
@v-dobrev OK.  Let's stick with what is there now (`target_link_libraries(... SCOREC::core)`).  If the other packages hit a similar issue we could reconsider using `target_link_libraries(... mfem)`.